packages/lesmis/src/api/queries/useEntityData.js

```diff
@@ -8,11 +8,19 @@ import { get } from '../api';
 
 export const useEntityData = entityCode => {
   const queryClient = useQueryClient();
-  return useQuery(['entity', entityCode], () => get(`entity/${entityCode}`), {
-    staleTime: 1000 * 60 * 60 * 1,
-    refetchOnWindowFocus: false,
-    // pre-poluate the query data from the cached entites data if it's available
-    initialData: () =>
-      queryClient.getQueryData('entities')?.find(entity => entity.code === entityCode),
-  });
+  return useQuery(
+    ['entity', entityCode],
+    () => {
+      return get(`entity/${entityCode}`);
+    },
+    {
+      staleTime: 1000 * 60 * 60 * 1,
+      refetchOnWindowFocus: false,
+      // pre-populate the query data from the cached entities data if it's available
+      initialData: () =>
+        queryClient
+          .getQueryData('entities')
+          ?.find(entity => entity.code === entityCode && !!entity.bounds), // check the entity exists and isn't "lean"/geo-data free
```

Comment:
Good catch - I remembered this yesterday and was going to mention it. It would be better if possible to solve this by relying on the cache keys. It would avoid us having to add this kind of logic in a few places and also it would ensure maximum use of the cache. In this instance, we are missing the cache benefit in places that just want the simple data (eg. the breadcrumbs)